tx/textual/internal/testdata/coin.json

```diff
@@ -0,0 +1,26 @@
+[
+  [{"amount": "1", "denom": "ucosm"}, { "denom": "COSM", "exponent": 6 }, "0.000001 COSM"],
+  [{"amount": "10", "denom": "ucosm"}, { "denom": "COSM", "exponent": 6 }, "0.00001 COSM"],
+  [{"amount": "100", "denom": "ucosm"}, { "denom": "COSM", "exponent": 6 }, "0.0001 COSM"],
+  [{"amount": "1000", "denom": "ucosm"}, { "denom": "COSM", "exponent": 6 }, "0.001 COSM"],
+  [{"amount": "10000", "denom": "ucosm"}, { "denom": "COSM", "exponent": 6 }, "0.01 COSM"],
+  [{"amount": "100000", "denom": "ucosm"}, { "denom": "COSM", "exponent": 6 }, "0.1 COSM"],
+  [{"amount": "1000000", "denom": "ucosm"}, { "denom": "COSM", "exponent": 6 }, "1 COSM"],
+  [{"amount": "10000000", "denom": "ucosm"}, { "denom": "COSM", "exponent": 6 }, "10 COSM"],
+  [{"amount": "1", "denom": "ucosm"}, { "denom": "COSM", "exponent": 2 }, "0.01 COSM"],
+  [{"amount": "10", "denom": "ucosm"}, { "denom": "COSM", "exponent": 2 }, "0.1 COSM"],
+  [{"amount": "100", "denom": "ucosm"}, { "denom": "COSM", "exponent": 2 }, "1 COSM"],
+  [{"amount": "1000", "denom": "ucosm"}, { "denom": "COSM", "exponent": 2 }, "10 COSM"],
+  [{"amount": "10000", "denom": "ucosm"}, { "denom": "COSM", "exponent": 2 }, "100 COSM"],
+  [{"amount": "100000", "denom": "ucosm"}, { "denom": "COSM", "exponent": 2 }, "1'000 COSM"],
+  [{"amount": "1000000", "denom": "ucosm"}, { "denom": "COSM", "exponent": 2 }, "10'000 COSM"],
+  [{"amount": "10000000", "denom": "ucosm"}, { "denom": "COSM", "exponent": 2 }, "100'000 COSM"],
+  [{"amount": "1", "denom": "point"}, { "denom": "POINT", "exponent": 0 }, "1 POINT"],
+  [{"amount": "10", "denom": "point"}, { "denom": "POINT", "exponent": 0 }, "10 POINT"],
+  [{"amount": "100", "denom": "point"}, { "denom": "POINT", "exponent": 0 }, "100 POINT"],
+  [{"amount": "1000", "denom": "point"}, { "denom": "POINT", "exponent": 0 }, "1'000 POINT"],
+  [{"amount": "10000", "denom": "point"}, { "denom": "POINT", "exponent": 0 }, "10'000 POINT"],
+  [{"amount": "100000", "denom": "point"}, { "denom": "POINT", "exponent": 0 }, "100'000 POINT"],
+  [{"amount": "1000000", "denom": "point"}, { "denom": "POINT", "exponent": 0 }, "1'000'000 POINT"],
+  [{"amount": "10000000", "denom": "point"}, { "denom": "POINT", "exponent": 0 }, "10'000'000 POINT"]
+]
```

Comment:
Add a test cases for

- zero amount
- coin without metadata
- metadata with entries other than display and coin names
- metadata without the display or coin name
- failures (see `timestamp.json` for examples)

Reply:
Added test cases for all items except "metadata with entries other than display and coin names". That would be a test-related json parsing error, which I don't think is really useful here?

Reply:
OK, I misunderstood, by `metadata with entries...` I understood fields in the metadata other than the usual `display` etc fields. I added.